services/cache/lru.go

```diff
@@ -182,6 +196,34 @@ func (l *LRU) get(key string, value interface{}) error {
 
 		l.evictList.MoveToFront(ent)
 
+		// We use a fast path for hot structs.
+		if msgpVal, ok := value.(msgp.Unmarshaler); ok {
+			_, err := msgpVal.UnmarshalMsg(e.value)
+			return err
+		}
```

Comment:
Is this actually needed?

Reply:
Fair concern. I did think about it, and at the end decided to keep it mainly for correctness. If somewhere in the codebase, we ever need to pass pointers to values for hot structs, this should still work.

0/5 if you prefer it to be removed, but we need to mention somewhere that passing pointers to values won't work for hot structs. I thought a couple of extra lines to not have any surprising behavior for a developer is a fair tradeoff.

Reply:
Sure, I guess it's fine to leave it also because it appears to be the proper way of dealing with cached objects.